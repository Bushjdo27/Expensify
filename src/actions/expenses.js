import {
    ADD_EXPENSE,
    ALL_EXPENSE,
    UPDATE_EXPENSE,
    DELETE_EXPENSE
} from './CONSTANT';


export const addExpense = ({id,description, note, amount, createdAt }) => {
    return {
        type: ADD_EXPENSE,
        expense: {
            id,
            description,
            note,
            amount,
            createdAt
        }
        

    }
}

export const updateExpense = (id, update) => {
    return {
        type: UPDATE_EXPENSE,
        id,
        expense: update

    }
}

export const deleteExpense = (id) => {
    return {
        type: DELETE_EXPENSE,
        id

    }
}