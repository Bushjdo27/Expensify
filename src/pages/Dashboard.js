import React, {Component} from 'react';
import {connect} from 'react-redux';
import ExpensesListItem from '../components/ExpensesListItem';
import selectExpenses from '../selector/expenses';
import ExpenseFilter from '../components/ExpenseFilter'
class Dashboard extends Component {
    render() {
        return (
            
            <div className="u--center">
                <main>
                    <ExpenseFilter />
                    <h1>Expense List</h1>
                    <ExpensesListItem list={this.props.expenses}/>
                </main>
            </div>
        )
        
    }
}

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(Dashboard);