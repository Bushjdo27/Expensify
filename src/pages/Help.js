import React, {Component} from 'react';

class Help extends Component {
    render() {
        return (
            <main>
                <p>This is Help Page</p>
            </main>
        )
        
    }
}


export default Help;