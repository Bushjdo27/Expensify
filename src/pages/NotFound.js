import React, {Component} from 'react';
import { Link } from 'react-router-dom'
const NotFound = () => {
    return (
        <main>
            <h1>
                Page Not Found
            </h1>
            <Link to="/">
                Go Home
            </Link>
        </main>
    )
}


export default NotFound;