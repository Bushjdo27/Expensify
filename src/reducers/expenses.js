import {
    ADD_EXPENSE,
    ALL_EXPENSE,
    UPDATE_EXPENSE,
    DELETE_EXPENSE
} from '../actions/CONSTANT';


export default function(state = [], action){

    switch(action.type) {
        case ALL_EXPENSE: 
            return state;
        case ADD_EXPENSE: 
            return [
                ...state,
                action.expense
            ];
        case UPDATE_EXPENSE: 
            return state.map(expense => {
                if(expense.id === action.expense.id){
                    return action.expense
                }
                return expense
            });
        case DELETE_EXPENSE: 
            return state.filter(expense => expense.id !== action.id);
        
        default: 
            return state;
    }
}