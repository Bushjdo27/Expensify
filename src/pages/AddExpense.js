import React, {Component} from 'react';
import ExpenseForm from '../components/ExpenseForm';
import {connect} from 'react-redux';
import {addExpense} from '../actions/expenses'

export class AddExpense extends Component {

    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/')
    }
    render() {
        return (
            <div className="u--center bg-grey">
                <main>
                    <h1>Add Expense</h1>
                    <ExpenseForm 
                        onSubmit={this.onSubmit}/>
                </main>
            </div>
        )
        
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addExpense: (expense) => dispatch(addExpense(expense))
    }
}

export default connect(undefined, mapDispatchToProps)(AddExpense);