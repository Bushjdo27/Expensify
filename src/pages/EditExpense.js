import React, {Component} from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm'
import {updateExpense, deleteExpense} from '../actions/expenses';


export class EditExpense extends Component {

    updateExpense = (expense)=>{
        this.props.updateExpense(this.props.expense.id, expense);
        this.props.history.push('/')
    }

    deleteExpense = ()=>{
        this.props.deleteExpense(this.props.expense.id);
        this.props.history.push('/')
    }
    
    render() {
        return (
            <div className="u--center">
                <main>
                    <p>This is Edit Page</p>
                    <ExpenseForm 
                        edit={true} 
                        expense={this.props.expense} 
                        onSubmit={this.updateExpense}/>

                    <button onClick={this.deleteExpense}>Remove</button>
                </main>
            </div>
            
        )
        
    }
}

const mapStateToProps = (state, props) =>{
    return {
        expense: state.expenses.find(item => item.id === props.match.params.id)
    }
}
const mapDispatchToProps = (dispatch, props) => {
    return {
        updateExpense: (id,expense) => dispatch(updateExpense(id, expense)),
        deleteExpense: (id) => dispatch(deleteExpense(id))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(EditExpense);