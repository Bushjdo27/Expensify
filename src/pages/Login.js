import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({Auth, startLogin}) => {
    return (
        <div>
            <h1>This is Login Page</h1>
            <button onClick={startLogin}>Login</button>
        </div>
    )
}

const mapStateToProps = (state) =>{
    return {
        //Auth: state.Auth
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin : () => dispatch(startLogin())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);