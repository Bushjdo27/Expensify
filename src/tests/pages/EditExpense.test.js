import React from 'react';
import { shallow } from 'enzyme';
import { EditExpense } from '../../pages/EditExpense';
import expenses from '../fixtures/expenses'
let updateExpenseSpy, deleteExpenseSpy, history, wrapper;

beforeEach(()=>{
    updateExpenseSpy = jest.fn();
    deleteExpenseSpy = jest.fn();
    history = {
        push: jest.fn()
    };
    wrapper = shallow(
        <EditExpense 
            updateExpense={updateExpenseSpy} 
            deleteExpense={deleteExpenseSpy} 
            history={history}
            expense={expenses[0]} 
        />);
});

test('should render EditExpense Page correctly',()=>{
    expect(wrapper).toMatchSnapshot();
});

test('should update expense correctly', ()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(updateExpenseSpy).toHaveBeenLastCalledWith(expenses[0].id, expenses[0])
});

test('should delete expense correctly',()=>{
    wrapper.find('button').simulate('click');
    expect(deleteExpenseSpy).toHaveBeenLastCalledWith(expenses[0].id)
})