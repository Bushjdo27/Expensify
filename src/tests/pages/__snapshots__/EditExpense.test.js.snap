// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense Page correctly 1`] = `
<div
  className="u--center"
>
  <main>
    <p>
      This is Edit Page
    </p>
    <ExpenseForm
      edit={true}
      expense={
        Object {
          "amount": 195,
          "createdAt": 0,
          "description": "Gum",
          "id": "1",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </main>
</div>
`;
